tools/clang/tools/dxcvalidator/CMakeLists.txt

```diff
@@ -9,6 +9,8 @@ set( LLVM_LINK_COMPONENTS
   DxilContainer
   DxilHash
   DxilValidation
+  DxilRootSignature
+  HLSL       # for gcc link error on AreDxilResourcesDense and DxilSignatureAllocator which are used in DxilValidation
```

Comment:
If these are the only dependencies on `HLSL.lib`, I think we could avoid this dependency with two things:
- `AreDxilResourcesDense`: this is only used to verify that resource IDs in metadata lists are sequential, starting at 0.  This could be locally checked in a simpler way for validation purposes.
- `DxilSignatureAllocator`: this is already implemented in a `.inl` file to make it easy to separate this dependency.

Reply:
AreDxilResourceDense should be easy to move to validation.
Where should we put DxilSignatureAllocator?
It is used in Validation and DxilPackSignatureElement.

Reply:
One option for `DxilSignatureAllocator.inl` would be to include it in a `.cpp` file under `dxildll` to provide an implementation when not linking in HLSL.lib.  Another solution might be to move the DxilPackSignatureElement stuff into the LLVMDXIL lib, or if we don't want to add it there, we could put it into a new lib instead.

Reply:
Created issue https://github.com/microsoft/DirectXShaderCompiler/issues/6898 to track this.